src/node/server/index.ts

```diff
@@ -123,8 +123,9 @@ export function createServer(config: ServerConfig): Server {
     if (optimizeDeps.auto !== false) {
       await require('../optimizer').optimizeDeps(config)
     }
-    context.port = port
-    return listen(port, ...args)
+    const listenResult = listen(port, ...args)
```

Comment:
Curious to understand why this needs to be an extra step @OriginalEXE. Wouldn't it be cleaner to just write:
```js
    context.port = server.address().port
    return listen(port, ...args)
```